src/web/page.rs

```diff
@@ -6,6 +6,27 @@ use iron::{IronResult, Set, status};
 use iron::response::Response;
 use handlebars_iron::Template;
 
+lazy_static! {
+    static ref RUSTC_RESOURCE_SUFFIX: String = load_rustc_resource_suffix()
+        .unwrap_or_else(|_| "???".into());
+}
+
+fn load_rustc_resource_suffix() -> Result<String, failure::Error> {
+    let conn = crate::db::connect_db()?;
+
+    let res = conn.query("SELECT value FROM config WHERE name = 'rustc_version';", &[])?;
+    if res.is_empty() {
+        failure::bail!("missing rustc version");
+    }
```

Comment:
It might be better to have this call `add_essential_files` instead of making it a manual step. You could move the logic you added above here to make sure it's not being called more than once.

Reply:
I'd prefer not to do that. In the future I'd like to see the frontend being a separate binary running on a separate server, instead of having both the frontend and the build daemon on the same machine. `add-essential-files` requires the build daemon, so if I make this change I would make splitting the two harder.

Reply:
Makes sense, I'd also like to see those split up at some point so the webserver isn't running as root.